naloge/maxvsota.tex

```diff
@@ -13,5 +13,88 @@
 \end{enumerate}
 \end{vprasanje}
 \begin{odgovor}
+
+\begin{enumerate}[(a)]
+
+\item Naj bo $V_i$ največja vsota nesosednjih členov zaporedja, ki jo lahko dobimo,
+če gledamo samo elemente od 1 do $i$, in vključuje $a_i$.
+\begin{align*}
+V_{-1} &= 0 \\
+V_0 &= 0 \\
+V_i &= \max_{j < i - 1}\left(V_j\right) + a_i
+\qquad (1 \le i \le n)
+\end{align*}
+Vidimo, da na $i$-tem koraku iščemo največjo prejšnjo nesosednjo vsoto $V_j$,
+ki ji bomo priključili element $a_i$, torej izračunamo največjo vsoto, 
+ki zadošča pogoju nesosednosti in vključuje člen $a_i$.
+
+Podamo lahko še alternativno rekurzivno zvezo za ta problem, in sicer 
+definirajmo $U_i$ kot največjo vsoto nesosednjih členov zaporadja, ki jo dobimo, 
+če gledamo samo elemente od 1 do $i$.
+Ta definicija se razlikuje od zgornje po tem,
+da dobljena vsota ne vključuje nujno elementa $a_i$.
+\begin{align*}
+U_{-1} &= 0 \\
+U_0 &= 0 \\
+U_i &= \max\left(U_{j - 2} + a_i, U_{j - 1}\right)
+\qquad (1 \le i \le n)
+\end{align*}
+Ideja tega rekurzivnega zapisa je podobna zgornjemu,
+le da se namesto iska\-nja maksimuma odloči,
+če bo $i$-ti člen vključila v vsoto oziroma ga bo izpustila.
+
+\item Algoritem bo sledil prvi rekurzivni zvezi, 
+maximum prejšnjih vsot pa bomo iskali na optimalen način, torej
+\begin{small}
+\begin{algorithmic}
+\Function{maxvsota}{$a$}
+	\State $n \gets \length(a)$
+	\State $\pred \gets [1, 2, \dots, n]$
+	\State $V \gets \operatorname{copy}(a)$
+	\State tmax $\gets 0$
+	\For{$i = 3, 4, \dots, n$}
+		\If{$V[i-2] > V$[tmax]}
+			\State tmax $\gets i - 2$
+		\EndIf
+		\If{$V$[tmax] $> 0$}
+			\State $V[i] \gets  V[i] + V$[tmax]
+			\State $\pred[i] \gets $tmax
+		\EndIf
+	\EndFor
+	\State tmax $\gets \argmax(V[$tmax$], V[n-1], V[n-2])$
```

Comment:
`\argmax` je smiselno uporabiti, če dejansko imaš argument (navadno v indeksu). Tukaj očitno hočeš indeks seznama *V*, kjer je dosežen maksimum, a ni to nikjer specificirano.